lib/globalize/active_record/instance_methods.rb

```diff
@@ -192,9 +196,9 @@ def save_translations!
       end
 
       def with_given_locale(attributes, &block)
-        attributes.symbolize_keys! if attributes.respond_to?(:symbolize_keys!)
+        symbolized_attributes = attributes.symbolize_keys if attributes.respond_to?(:symbolize_keys)
```

Comment:
@parndt 
Thank you very much for your comment! I totally agree with your opinion.
I followed your suggestion and optimised my code!